File: backend/services/data-collection-service/src/main/java/com/datamate/collection/domain/model/entity/CollectionTask.java
package com.datamate.collection.domain.model.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.datamate.collection.common.enums.SyncMode;
import com.datamate.collection.common.enums.TaskStatus;
import com.datamate.collection.common.enums.TemplateType;
import com.datamate.common.domain.model.base.BaseEntity;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.Collections;
import java.util.Map;
import java.util.UUID;

/**
 * 数据采集任务实体（与数据库表 t_dc_collection_tasks 对齐）
 */
@Getter
@Setter
@TableName(value = "t_dc_collection_tasks", autoResultMap = true)
public class CollectionTask extends BaseEntity<String> {
    private String name;
    private String description;
    private TemplateType taskType; // 任务类型
    private String targetPath; // 目标存储路径
    private String config; // DataX JSON 配置，包含源端和目标端配置信息
    private TaskStatus status;
    private SyncMode syncMode; // ONCE / SCHEDULED
    private String scheduleExpression;
    private Integer retryCount;
    private Integer timeoutSeconds;
    private Long maxRecords;
    private String sortField;
    private String lastExecutionId;

    public void addPath() {
        try {
            ObjectMapper objectMapper = new ObjectMapper();
            Map<String, Object> parameter = objectMapper.readValue(
                config,
                new TypeReference<>() {}
            );
            parameter.put("destPath", "/dataset/local/" + id);
            parameter.put("filePaths", Collections.singletonList(parameter.get("destPath")));
            config = objectMapper.writeValueAsString(parameter);
        } catch (JsonProcessingException e) {
            throw new RuntimeException(e);
        }
    }

    public void initCreateParam() {
        this.id = UUID.randomUUID().toString();
        this.targetPath = "/dataset/local/" + id;
        this.status = TaskStatus.READY;
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
    }
}

File: runtime/datamate-python/app/models/collection/collection_task.py
from sqlalchemy import Column, String, Text, Integer, BigInteger, TIMESTAMP
from sqlalchemy.sql import func
from app.db.database import Base

class CollectionTask(Base):
    """数据归集任务模型"""
    
    __tablename__ = "t_dc_collection_tasks"
    
    id = Column(String(36), primary_key=True, comment="任务ID（UUID）")
    name = Column(String(255), nullable=False, comment="任务名称")
    description = Column(Text, nullable=True, comment="任务描述")
    sync_mode = Column(String(20), default='ONCE', comment="同步模式：ONCE/SCHEDULED")
    config = Column(Text, nullable=False, comment="归集配置（DataX配置），包含源端和目标端配置信息")
    schedule_expression = Column(String(255), nullable=True, comment="Cron调度表达式")
    status = Column(String(20), default='DRAFT', comment="任务状态：DRAFT/READY/RUNNING/SUCCESS/FAILED/STOPPED")
    retry_count = Column(Integer, default=3, comment="重试次数")
    timeout_seconds = Column(Integer, default=3600, comment="超时时间（秒）")
    max_records = Column(BigInteger, nullable=True, comment="最大处理记录数")
    sort_field = Column(String(100), nullable=True, comment="增量字段")
    last_execution_id = Column(String(36), nullable=True, comment="最后执行ID（UUID）")
    created_at = Column(TIMESTAMP, server_default=func.current_timestamp(), comment="创建时间")
    updated_at = Column(TIMESTAMP, server_default=func.current_timestamp(), onupdate=func.current_timestamp(), comment="更新时间")
    created_by = Column(String(255), nullable=True, comment="创建者")
    updated_by = Column(String(255), nullable=True, comment="更新者")
    
    def __repr__(self):
        return f"<CollectionTask(id={self.id}, name={self.name}, status={self.status})>"
